mender-deb-package

```diff
@@ -0,0 +1,107 @@
+#/bin/bash
+
+set -ex
+
+show_help_and_exit() {
+  cat << EOF
+
+Usage: $0 mender-version
+
+NOTE: The script expects an /output directory where to store the generated packages. If
+running it from a container, create a volume for such directory
+
+EOF
+  exit 1
+}
+
+version=""
+
+while (( "$#" )); do
+  case "$1" in
+    -*)
+      echo "Error: unsupported option $1"
+      show_help_and_exit_error
+      ;;
+    *)
+      version="$1"
+      shift
+      ;;
+  esac
+done
+
+if [ -z "$version"]; then
+	show_help_and_exit
+fi
+
+if [ ! -d "/output" ]; then
+  echo "Error: /output directory doesn't exist"
+  show_help_and_exit
+fi
+
+#Print misc env variables for debug
+echo "WORKDIR="`pwd`
+echo "PATH="$PATH
+echo "CC="$CC
+echo "GOPATH="$GOPATH
+echo "version=$version"
+
+mkdir debian && mkdir debian/source
+
+#TODO: find a way to take this from the Changelog itself
+dch --create -v ${version}-1 --package mender "Release ${version}. See online docs for Changelog"
+
+echo "10" > debian/compat
+
+echo "3.0 (quilt)" > debian/source/format
+
+cat << EOF > debian/control
+Source: mender
+Maintainer: Mender Team <menderall@northern.tech>
+Section: misc
+Priority: optional
+Standards-Version: 3.9.2
+Build-Depends: debhelper (>= 9)
+
+Package: mender
+Architecture: armhf
+Depends: \${shlibs:Depends}, \${misc:Depends}
+Description: mender client
+ Mender is an open source over-the-air (OTA) software updater for embedded Linux devices.
+EOF
+
+cp LICENSE debian/copyright
+
+cat << EOF > debian/rules
+#!/usr/bin/make -f
+
+%:
+	dh \$@
+
+override_dh_auto_build:
+	env CGO_CFLAGS="-I/usr/local/include" CGO_LDFLAGS="-L/usr/local/lib" \
+	GOOS=linux GOARCH=arm GOARM=7 CGO_ENABLED=1 \
```

Comment:
@kacf I changed GOARM to 7. Is that correct? Should we just not specify it and go with the Golang default? On the Raspberry Pi 3 (ARMv7), it works for the three versions (5,6,7).

Reply:
Yes, it works on RPi because ARMv7 is backwards compatible with ARMv5 and 6, but not vice versa. So an ARMv5 device will not be able to run ARMv7 binaries. Hence why I think we should compile with that.

Reply:
And I also made experiments setting GOARM=5, =6, =7... When I read back the binaries, I always get the same result, that it is compiled for v7. See below the output of readelf from a build with GOARM=5

# readelf -A mender_arm5

File: mender_arm5
Attribute Section: aeabi
File Attributes
  Tag_CPU_name: "7-A"
  Tag_CPU_arch: v7
  Tag_CPU_arch_profile: Application
  Tag_ARM_ISA_use: Yes
  Tag_THUMB_ISA_use: Thumb-2
  Tag_FP_arch: VFPv3-D16
  Tag_ABI_PCS_wchar_t: 4
  Tag_ABI_FP_rounding: Needed
  Tag_ABI_FP_denormal: Needed
  Tag_ABI_FP_exceptions: Needed
  Tag_ABI_FP_number_model: IEEE 754
  Tag_ABI_align_needed: 8-byte
  Tag_ABI_enum_size: int
  Tag_ABI_VFP_args: VFP registers
  Tag_CPU_unaligned_access: v6

Reply:
I leave it with GOARM=5, though

Reply:
Yes, but I think this is because we use `CGO_ENABLED=1`. It is GCC that does the linking, not Golang, so it boils down to whether GCC supports it. The actual code that Golang generates may still be ARMv5, but that doesn't help if the binary is wrapped in an ARMv7 header.

Reply:
Makes sense! Thank you for the clarification.